### PR TITLE
1st pr from dev to main: landing page, navbar, footer, slider, viewing page, DB

### PR DESCRIPTION
**Pull request by FabienRollet**

### CODE REVIEW

File: client/.gitignore
Comment (FranckWebPro) [inline client/.gitignore:1]: style.html written twice

File: client/package.json [hunk 1]
```diff
@@ -7,21 +7,27 @@
     "preview": "vite preview"
   },
   "dependencies": {
+    "fluid-tailwind": "^1.0.0",
```
Comment (FranckWebPro) [inline client/package.json]: fluid tailwind to remove from dependencies

File: client/src/components/Footer.jsx [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+export default function Footer() {
+  return (
+    <footer className="bg-[var(--primaryColor)] p-4 lg:flex lg:p-10 ">
+      <h2 className="italic font-extrabold uppercase">
+        <Link to="/">Sweat Stream</Link>
+      </h2>
+      <ul className="flex flex-wrap gap-x-6 gap-y-6 lg:grow lg:justify-evenly lg:leading-10">
+        <li className="grow lg:grow-0">
+          <h3>
+            <Link to="/">Pages</Link>
```
Comment (FranckWebPro) [inline client/src/components/Footer.jsx]: Pages should not be a link

File: client/src/components/Footer.jsx [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+              <Link to="/">Inscription coach</Link>
+            </li>
+            <li>
+              <Link to="/">Connexion</Link>
+            </li>
+          </ul>
+        </li>
+        <li className="grow lg:grow-0">
+          <h3>
+            <Link to="/">Catégories</Link>
```
Comment (FranckWebPro) [inline client/src/components/Footer.jsx]: Catégories should not be a link

File: client/src/components/Header.jsx [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    lg:static lg:max-h-16 lg:flex-row lg:justify-around lg:opacity-100 lg:transform-none
+    ${isMenuOpen ? "opacity-100 transform translate-y-0" : "-translate-y-[100vh]"}`;
+
+  const burgerButtonClasses = `burgerMenu relative w-10 h-7 bg-transparent text-0 hover:bg-transparent none border-none 
+    flex justify-self-end text-[0]
+    lg:hidden
+    ${isMenuOpen ? "active" : ""}`;
+
+  const objectiveSectionClasses = `absolute top-[4.5rem] left-0 right-0 bottom-0 w-full 
+    uppercase bg-white list-none justify-around items-center transform transition-all duration-300 text-center
```
Comment (Sylvainlfe) [inline client/src/components/Header.jsx:37]: uppercase non fonctionnel - uppercase dans button objectif

File: client/src/components/Header.jsx [hunk 5]
```diff
@@ -0,0 +1,74 @@
+import { useState, useEffect } from "react";
+import Navbar from "./Navbar";
+
+function Header() {
+  const [isMenuOpen, setIsMenuOpen] = useState(false);
```
Comment (FranckWebPro) [inline client/src/components/Header.jsx]: Add mobile for the variable names

File: client/src/components/Header.jsx [hunk 6]
```diff
@@ -0,0 +1,74 @@
+import { useState, useEffect } from "react";
+import Navbar from "./Navbar";
+
+function Header() {
+  const [isMenuOpen, setIsMenuOpen] = useState(false);
+  const [isObjectivesMenuOpen, setIsObjectivesMenuOpen] = useState(false);
+
+  const closeMenu = () => {
```
Comment (FranckWebPro) [inline client/src/components/Header.jsx:8]: same

File: client/src/components/Navbar.jsx [hunk 7]
```diff
@@ -0,0 +1,159 @@
...
+            alt="retour à l'accueil"
+          />
+        </Link>
+        <button
+          type="button"
+          onClick={handleClick}
+          className={burgerButtonClasses}
+          title="Menu déroulant"
+        >
+          Menu déroulant
```
Comment (FranckWebPro) [inline client/src/components/Navbar.jsx]: remove "menu deroulant"

File: client/src/pages/SubscriptionPage.jsx [hunk 8]
```diff
@@ -0,0 +1,142 @@
...
+                type={info.type}
+                id={info.id}
+                name={info.text}
+                value={formValues[info.id]}
+                onChange={handleChange}
+                className="border-b-2 border-dc py-1 focus:border-b-2 focus:border-pc transition-colors focus:outline-none peer bg-inherit w-full"
+              />
+              <label
+                htmlFor={info.id}
+                className={`left-1 top-0 font-semibold cursor-text absolute transition-all
```
Comment (Sylvainlfe) [inline client/src/pages/SubscriptionPage.jsx]: fix défaut de la logique pour garder le label en -top-4 quand il y a du texte dans le champ

File: server/app/config.js
Comment (FranckWebPro) [inline server/app/config.js:1]: remove comment ?

**Review by Sylvainlfe — APPROVED** [review]

**Review by FranckWebPro — APPROVED** [review]

File: client/src/index.css [hunk 9]
```diff
@@ -0,0 +1,224 @@
...
+
+* {
+  box-sizing: border-box;
+  margin: 0;
+  padding: 0;
+}
+
+body {
+  font-family: var(--textFont);
+  font-size: 1rem;
```
Comment (Yoann-Fortin) [inline client/src/index.css]: Dead code

File: client/src/index.css [hunk 10]
```diff
@@ -0,0 +1,224 @@
...
+
+li {
+  margin-left: 0.25rem;
+}
+
+label {
+  font-family: var(--textFont);
+}
+
+label:focus {
```
Comment (Yoann-Fortin) [inline client/src/index.css]: Dead code

File: client/src/components/Header.jsx [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+    setIsMobileMenuOpen(!isMobileMenuOpen);
+  };
+
+  const handleClickObjectivesMenu = () => {
+    setIsObjectivesMenuOpen(!isObjectivesMenuOpen);
+  };
+
+  useEffect(() => {
+    if (isMobileMenuOpen) {
+      document.body.classList.add("overflow-hidden");
```
Comment (Yoann-Fortin) [inline client/src/components/Header.jsx]: toggle is better

File: client/src/components/Header.jsx [hunk 12]
```diff
@@ -0,0 +1,74 @@
...
+    } else {
+      document.body.classList.remove("overflow-hidden");
+    }
+  }, [isMobileMenuOpen]);
+
+  const signInClasses = `
```
Comment (Yoann-Fortin) [inline client/src/components/Header.jsx]: Too much code duplication

File: client/src/components/HeroSection.jsx [hunk 13]
```diff
@@ -0,0 +1,30 @@
+import { Link } from "react-router-dom";
+
+export default function HeroSection() {
+  return (
+    <header
```
Comment (Yoann-Fortin) [inline client/src/components/HeroSection.jsx]: Wrong uses this tag

File: client/src/components/Navbar.jsx [hunk 14]
```diff
@@ -0,0 +1,158 @@
...
+  signUpClasses,
+  burgerButtonClasses,
+  menuListClasses,
+  objectivesButtonClasses,
+  objectiveSectionClasses,
+}) {
+  return (
+    <header className="bg-[var(--secondaryColor)] h-20 lg:h-24 flex items-center">
+      <nav
+        id="navbar"
```
Comment (Yoann-Fortin) [inline client/src/components/Navbar.jsx]: no justify

File: client/src/components/Slider.jsx [hunk 15]
```diff
@@ -0,0 +1,78 @@
...
+import "../assets/styles/slider.css";
+import { Navigation, Pagination } from "swiper/modules";
+import { Link } from "react-router-dom";
+import image1 from "../assets/images/musculation.jpg";
+import image2 from "../assets/images/yoga.jpg";
+import image3 from "../assets/images/nutrition.jpg";
+import image4 from "../assets/images/pilat.jpg";
+import image5 from "../assets/images/training.jpg";
+
+const sportList = [
```
Comment (Yoann-Fortin) [inline client/src/components/Slider.jsx]: This logic must be transmitted by the parent

File: client/src/components/infoSection.jsx [hunk 16]
```diff
@@ -0,0 +1,22 @@
+export default function registerSection() {
+  return (
+    <section
```
Comment (Yoann-Fortin) [inline client/src/components/infoSection.jsx]: no justify

File: client/src/components/registerSection.jsx [hunk 17]
```diff
@@ -0,0 +1,35 @@
+import { Link } from "react-router-dom";
+
+export default function registerSection() {
+  return (
+    <section
```
Comment (Yoann-Fortin) [inline client/src/components/registerSection.jsx]: no justify

File: client/src/index.css [hunk 18]
```diff
@@ -0,0 +1,224 @@
...
+}
+
+body {
+  font-family: var(--textFont);
+  font-size: 1rem;
+  color: var(--darkColor);
+  background-color: var(--lightColor);
+}
+p {
+  font-size: 1rem;
```
Comment (Yoann-Fortin) [inline client/src/index.css]: Dead code

File: client/src/index.css [hunk 19]
```diff
@@ -0,0 +1,224 @@
...
+button {
+  height: 2.5rem;
+  padding: 0.5rem 1rem;
+  text-align: center;
+  border-radius: 15px;
+  font-weight: 600;
+  background-color: var(--primaryColor);
+  outline: none;
+  border: none;
+  transition: all ease 300ms;
```
Comment (Yoann-Fortin) [inline client/src/index.css:103]: You'll need to disable the animations

File: client/src/index.css [hunk 20]
```diff
@@ -0,0 +1,224 @@
...
+
+button:hover {
+  background-color: var(--primaryDark);
+  color: var(--lightColor);
+}
+
+a {
+  color: var(--primaryDark);
+  text-decoration: none;
+  transition: all ease 300ms;
```
Comment (Yoann-Fortin) [inline client/src/index.css:115]: same

File: client/src/index.css [hunk 21]
```diff
@@ -0,0 +1,224 @@
...
+}
+
+.center {
```
Comment (Yoann-Fortin) [inline client/src/index.css]: Dead code

File: client/src/index.css [hunk 22]
```diff
@@ -0,0 +1,224 @@
...
+  width: 50%;
+  margin: auto;
+  display: flex;
+  flex-direction: column;
+  align-items: flex-start;
+  justify-content: center;
+  gap: 1rem;
+}
+
+.img-shadow {
```
Comment (Yoann-Fortin) [inline client/src/index.css]: Dead code

File: client/src/index.css [hunk 23]
```diff
@@ -0,0 +1,224 @@
...
+  }
+
+  footer {
+    padding: 1rem 3rem;
+  }
+}
+
+/* Burger-design */
+
+.line-burger {
```
Comment (Yoann-Fortin) [inline client/src/index.css]: config tailwind

File: client/src/App.jsx [hunk 24]
```diff
@@ -1,88 +1,18 @@
...
-          rel="noopener noreferrer"
-        >
-          Wild Code School
-        </a>
-      </footer>
-    </main>
+  return (
+    <>
+      <Header />
+      <main>
```
Comment (Yoann-Fortin) [inline client/src/App.jsx]: Two main ?

File: client/src/pages/HomePage.jsx [hunk 25]
```diff
@@ -0,0 +1,18 @@
+import HeroSection from "../components/HeroSection";
+import RegisterSection from "../components/registerSection";
+import InfoSection from "../components/infoSection";
+import Slider from "../components/Slider";
+
+function Home() {
+  return (
+    <main>
+      <HeroSection />
+      <section className="pt-24 pb-16 md:pt-32">
```
Comment (Yoann-Fortin) [inline client/src/pages/HomePage.jsx]: why ?

File: client/src/pages/SubscriptionPage.jsx
Comment (Yoann-Fortin) [inline client/src/pages/SubscriptionPage.jsx:1]: Separate logic from display

File: client/tailwind.config.js
Comment (Yoann-Fortin) [inline client/tailwind.config.js:1]: wrong names

File: server/app/config.js [hunk 26]
```diff
@@ -12,8 +12,9 @@ const app = express();
 
 // CORS (Cross-Origin Resource Sharing) is a security mechanism in web browsers that blocks requests from a different domain than the server.
 // You may find the following magic line in forums:
+const cors = require("cors");
 
-// app.use(cors());
+app.use(cors());
```
Comment (Yoann-Fortin) [inline server/app/config.js]: other line to decomment

**Review by Yoann-Fortin — CHANGES_REQUESTED** [review]

File: client/src/components/SignIn.jsx [hunk 27]
```diff
@@ -0,0 +1,60 @@
+import { Link } from "react-router-dom";
+import PropTypes from "prop-types";
+import fitnessImg from "../assets/images/training.jpg";
+import "../assets/styles/form.css";
+
+function SignIn({ handleChangeScribe, fields, formValues, generateFieldLabelClass }) {
+  return (
+    <section className="flex justify-center items-center gap-10 ">
+      <figure className="hidden lg:flex justify-center">
```
Comment (Yoann-Fortin) [inline client/src/components/SignIn.jsx]: The figure tag is generally used in conjunction with the figcaption tag.

File: client/src/components/SignIn.jsx [hunk 28]
```diff
@@ -0,0 +1,60 @@
...
+  return (
+    <section className="flex justify-center items-center gap-10 ">
+      <figure className="hidden lg:flex justify-center">
+        <img
+          src={fitnessImg}
+          alt="sportifs"
+          className="img-shadow rounded-xl mb-4 object-cover h-96 "
+        />
+      </figure>
+      <article className="w-96">
```
Comment (Yoann-Fortin) [inline client/src/components/SignIn.jsx]: This article tag is used for styling, I think.

File: client/src/components/Slider.jsx [hunk 29]
```diff
@@ -0,0 +1,60 @@
...
+      {sportList.map((sport) => (
+        <SwiperSlide key={sport.name}>
+          <Link to="/category">
+            <img
+              src={sport.imgSrc}
+              alt={sport.name}
+              className="img-shadow w-[calc(100%-15px)] mr-auto rounded-xl mb-4 h-60 object-cover"
+            />
+          </Link>
+          <p className="flex justify-center items-center font-bold">
```
Comment (Yoann-Fortin) [inline client/src/components/Slider.jsx]: It's strange not to be able to click on the text but only on the image. Something to think about

File: client/src/components/Subscription.jsx [hunk 30]
```diff
@@ -0,0 +1,101 @@
...
+  handleChange,
+  fields,
+  formValues,
+  customerButton,
+  professionalButton,
+  generateFieldLabelClass,
+}) {
+  return (
+    <section className="flex justify-center items-center gap-10 ">
+      <figure className="hidden lg:flex items-center justify-center">
```
Comment (Yoann-Fortin) [inline client/src/components/Subscription.jsx]: See comment above

File: client/src/index.css [hunk 31]
```diff
@@ -0,0 +1,184 @@
...
+}
+
+button {
+  height: 2.5rem;
+  padding: 0.5rem 1rem;
+  text-align: center;
+  border-radius: 15px;
+  font-weight: 600;
+  background-color: var(--primaryColor);
+  outline: none;
```
Comment (Yoann-Fortin) [inline client/src/index.css]: Disabling the outline property without managing focus disables the default keyboard navigation behavior to indicate that this element has taken focus. Either leave the default behavior, or manage to signal focus

File: client/src/pages/SubscriptionPage.jsx [hunk 32]
```diff
@@ -0,0 +1,108 @@
...
+    setFormValues({
+      firstname: "",
+      lastname: "",
+      email: "",
+      password: "",
+      siret: "",
+    });
+  };
+
+  const handleClickProfessional = () => {
```
Comment (Yoann-Fortin) [inline client/src/pages/SubscriptionPage.jsx]: Duplicate code

**Review by Yoann-Fortin — CHANGES_REQUESTED** [review]

File: client/src/components/Slider.jsx [hunk 33]
```diff
@@ -0,0 +1,60 @@
...
+    >
+      {sportList.map((sport) => (
+        <SwiperSlide key={sport.name}>
+          <Link to="/category">
+            <img
+              src={sport.imgSrc}
+              alt={sport.name}
+              className="img-shadow w-[calc(100%-15px)] mr-auto rounded-xl mb-4 h-60 object-cover"
+            />
+          <p className="flex justify-center items-center font-bold">
```
Comment (Yoann-Fortin) [inline client/src/components/Slider.jsx]: A p tag cannot be inside an a tag

File: client/src/pages/SubscriptionPage.jsx [hunk 34]
```diff
@@ -0,0 +1,102 @@
...
+  email: "",
+  password: "",
+  siret: "",
+};
+
+function SubscriptionPage() {
+  const [fields, setFields] = useState(textLabel);
+  const [formValues, setFormValues] = useState(emptyFields);
+
+  const handleClickCustomer = () => {
```
Comment (Yoann-Fortin) [inline client/src/pages/SubscriptionPage.jsx]: Code is always duplicated between handleClickCustomer and handleClickProfessional

**Review by Yoann-Fortin — CHANGES_REQUESTED** [review]

**Review by Sylvainlfe — APPROVED** [review]

**Review by Yoann-Fortin — APPROVED** [review]

**Pull request merged**